tmt/export.py

```diff
@@ -0,0 +1,182 @@
+# coding: utf-8
+
+""" Export metadata into nitrate """
+
+from click import echo, style
+
+import subprocess
+import tmt.utils
+import nitrate
+import pprint
+import email
+import fmf
+import os
+
+log = fmf.utils.Logging('tmt').logger
+# Needed for nitrate.Component
+DEFAULT_PRODUCT = nitrate.Product(name='RHEL Tests')
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#  Export
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+def export_to_nitrate():
+    """ Export fmf metadata to nitrate test cases """
+    path = os.path.abspath('.')
+    echo(style("Checking the '{0}' directory.".format(path), fg='red'))
+
+    tree = fmf.Tree(path)
+    for fmf_case in list(tree.prune(names=[path[len(tree.root):]])):
```

Comment:
Nitpick: list is not necessary, prune() is iterable

Reply:
I would recommend removing the for loop altogether. Iterating over (selected) tests is already done in the `export()` method under `tmt/cli.py`. In this way you get filtering for free and it would be more consistent:

    tmt test export /tests/docs
    tmt test export --filter tier:1

Also one more thought related to the way how we choose the export target: What about using the already present option `--format` for that? It could look like this:

    tmt test export --format nitrate

Then it would be easy to update the `Test.export()` method to call `export_to_nitrate()`. What do you think?

Reply:
Second part fixed by 1ce3c26
I haven't implemented first part yet. How do you iterate over the selected tests?

Reply:
Just make use of the [existing loop](https://github.com/psss/tmt/blob/master/tmt/cli.py#L484) (no need to change there anything) and call the new nitrate export implementation from [Test.export()](https://github.com/psss/tmt/blob/master/tmt/base.py#L176) method.

Reply:
Partially fixed by 2509a2c
I didn't move the calling of export_to_nitrate() into Test.export() because I found it to be more confusing than calling it from tmt.cli.export(). If you think it's better to call it from Test.export() then move it there please (I couldn't figure out how to naturally add there).